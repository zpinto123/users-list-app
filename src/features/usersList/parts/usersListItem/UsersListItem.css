.row {
  height: 35px;
  justify-content: center;
  align-items: center;
  text-align: center;
  background-color: #71b4ce;
  margin: 1px;
  border-radius: 5px;
}

.row:hover {
  background-color: #71b4ee;
}

.selected {
  background-color: #71b4ee;
}

.userDisplayName {
  color: white;
  text-align: center;
  font-weight: 600;
  line-height: 35px;
}

.userDetailsContainer {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: space-between;
  height: 130px;
  border-radius: 5px;
  background-color: #76a3b5;
}

.userInfoContainer {
  display: flex;
  flex-direction: column;
  flex: 0.8;
  justify-content: space-evenly;
  align-items: space-evenly;
}

.userAvatarContainer {
  flex: 0.3;
  justify-content: center;
  align-items: center;
  margin: auto;
}

.userAvatar {
  height: 50px;
  width: 50px;
}

.label {
  color: white;
  font-weight: 600;
  margin-right: 7px;
}

.value {
  text-align: left;
  color: white;
}

.userInfo {
  display: flex;
  justify-content: flex-start;
}
