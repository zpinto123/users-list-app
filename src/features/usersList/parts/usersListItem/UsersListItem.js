import React, { Fragment } from "react";
import PropTypes from "prop-types";

import UserDetails from "../userDetails";

import "./UsersListItem.css";

const UsersListItem = ({ userInfo, isSelected, handleClick }) => (
  <Fragment>
    <div
      className={`row ${isSelected ? "selected" : ""}`}
      onClick={handleClick}
    >
      <label className="userDisplayName">{userInfo.displayName}</label>
    </div>
    {isSelected && <UserDetails userInfo={userInfo} />}
  </Fragment>
);

UsersListItem.propTypes = {
  userInfo: PropTypes.shape({
    name: PropTypes.string.isRequired,
    surname: PropTypes.string.isRequired,
    email: PropTypes.string,
    employer: PropTypes.string
  }),
  handleClick: PropTypes.func.isRequired,
  isSelected: PropTypes.bool
};

UsersListItem.defaultProps = {
  isSelected: false
};

export default UsersListItem;
