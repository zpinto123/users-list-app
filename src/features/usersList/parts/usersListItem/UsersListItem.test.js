import React from "react";
import renderer from "react-test-renderer";

import UsersListItem from "./UsersListItem";

describe("<UsersListItem />", () => {
  it("renders correctly", () => {
    const props = {
      userInfo: {
        displayName: "José Pinto",
        name: "José Pinto",
        surname: "Pinto",
        email: "super@super.com",
        employer: "JpanOTECH"
      },
      handleClick: jest.fn()
    };

    const component = renderer.create(<UsersListItem {...props} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
