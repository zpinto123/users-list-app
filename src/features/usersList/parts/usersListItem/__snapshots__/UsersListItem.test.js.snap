// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UsersListItem /> renders correctly 1`] = `
<div
  className="row "
  onClick={[MockFunction]}
>
  <label
    className="userDisplayName"
  >
    Jos√© Pinto
  </label>
</div>
`;
